lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -2511,16 +2510,27 @@ LogicalResult ConvertAtenOp<AtenFlattenUsingIntsOp>::matchAndRewrite(
     return rewriter.notifyMatchFailure(op,
                                        "end_dim must be larger than start_dim");
 
+  SmallVector<int64_t> oldShape;
+
+  if (selfType.hasStaticShape()) {
+    for (auto s : selfType.getShape())
+      oldShape.push_back(s);
+  } else {
+    // Pushing unknown sizes
+    for (unsigned i = 0; i < selfRank; i++)
+      oldShape.push_back(ShapedType::kDynamic);
+  }
```

Comment:
I'm a bit confused by this. `selfType` will already contain `ShapedType::kDynamic` for the dimensions that are dynamic. Why do we need to wipe out all the static information?

Reply:
I guess I haven't considered the case I will quickly make the fix.